elb_log_ingestor/elb_log_fetcher.py

```diff
@@ -0,0 +1,152 @@
+"""
+Retrieves ELB log files
+"""
+import io
+import logging
+import pathlib
+import queue
+
+logger = logging.Logger(__self__)
+
+
+class S3LogFetcher:
+    """
+    Fetches logs from S3, moves logs in S3 around to indicate they're processing/processed
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        s3_client: "boto3.resources.factory.s3.ServiceResource",
+        unprocessed_prefix: str,
+        processing_prefix: str,
+        processed_prefix: str,
+        to_do: queue.Queue,
+        done: queue.Queue,
+        start_queue_size: int = 5,
```

Comment:
The platform logs at like 600 records a second, I would assume ELB would log at a similar rate, so it might be better to have a larger starting bucket. I have no idea how fast the ELB logs but mostly food for thought.